.github/workflows/plugin-update.yml

```diff
@@ -0,0 +1,117 @@
+        with:
+          cache: false # save cache space for vault builds: https://github.com/hashicorp/vault/pull/21764
+          go-version-file: .go-version
+
+      - name: update plugin
+        run: |
+          go get "github.com/${{ inputs.repo }}@${{ inputs.plugin_tag }}"
+          go mod tidy
+
+      - name: detect changes
+        run: |
+          count=$(git status --porcelain=v1 2>/dev/null | wc -l)
+          if [ "$count" -eq 0 ]; then
+            echo "error: no updates were made for repo ${{ inputs.repo }} with tag ${{ inputs.plugin_tag }}"
+            exit 1
+          fi
+
+      - name: commit/push
+        run: |
+          git config user.name hc-github-team-secure-vault-ecosystem
+          git config user.email hc-github-team-secure-vault-ecosystem@users.noreply.github.com
+          git add .
+          git commit -m "Automated dependency upgrades"
+          git push -f origin ${{ github.ref_name }}:"$VAULT_BRANCH"
+
+      - name: Open pull request if needed
+        id: pr
+        env:
+          GITHUB_TOKEN: ${{secrets.ELEVATED_GITHUB_TOKEN}}
+        # Only open a PR if the branch is not attached to an existing one
+        run: |
+          PR=$(gh pr list --head "$VAULT_BRANCH" --json number -q '.[0].number')
+
+          if [ -z "$PR" ]; then
+            gh pr create \
+            --head "$VAULT_BRANCH" \
+            --title "Update ${{ inputs.repo }} to ${{ inputs.plugin_tag }}" \
+            --body "Updates ${{ inputs.repo }} to verify vault CI. Full log: https://github.com/hashicorp/vault/actions/runs/${{ github.run_id }}"
+
+            echo "vault_pr_num=$(gh pr list --head "$VAULT_BRANCH" --json number -q '.[0].number')" >> "$GITHUB_OUTPUT"
+            echo "vault_pr_url=$(gh pr list --head "$VAULT_BRANCH" --json url -q '.[0].url')" >> "$GITHUB_OUTPUT"
+          else
+            echo "Pull request already exists, won't create a new one."
+          fi
+
+      - name: Add changelog
+        continue-on-error: true
+        run: |
+          if [ -z "$vault_pr_num" ]; then
+            echo "error: no vault PR found"
+            exit 1
+          fi
+
+          # strip "hashicorp/" from repo name
+          PLUGIN=$(echo ${{ inputs.repo }} | awk -F/ '{print $NF}')
+          echo "plugin: $PLUGIN"
+
+          # plugin type is one of auth/secrets/database
+          PLUGIN_TYPE=$(echo "$PLUGIN" | awk -F- '{print $3}')
+          echo "plugin type: $PLUGIN_TYPE"
+
+          # plugin service is the last element in the repo name
+          PLUGIN_SERVICE=$(echo "$PLUGIN" | awk -F- '{print $4}')
+          echo "plugin service: $PLUGIN_SERVICE"
+
+          echo "\`\`\`release-note:change
+          ${PLUGIN_TYPE}/${PLUGIN_SERVICE}: Update plugin to ${{ inputs.plugin_tag }}
+          \`\`\`" > "changelog/${vault_pr_num}.txt"
+
+          git add .
+          git commit -m "Add changelog"
+          git push origin ${{ github.ref_name }}:"$VAULT_BRANCH"
+
+      - name: Add labels to Vault PR
```

Comment:
Would it be possible to detect & configure what is the upcoming Vault milestone so a release manager would pick up any straggling plugin update PRs if we miss some?

Reply:
Maybe? I considered this but didn't spend any time on it. This would be a great improvement though